### PR TITLE
Add support for optimal DFA on LLVM bitcode

### PR DESCRIPTION
**Pull request by zhengyangl**
- backward analysis (Demanded bits) is not supported

### CODE REVIEW

File: lib/Tool/CandidateMapUtils.cpp [hunk 1]
```diff
@@ -43,6 +72,18 @@ void souper::AddModuleToCandidateMap(InstContext &IC, ExprBuilderContext &EBC,
   }
 }
 
+static bool isInferDFA() {
+  return InferNeg || InferNonNeg || InferKnownBits || InferPowerTwo ||
+         InferNonZero || InferSignBits || InferRange;
+}
+
+static std::string convertToStr(bool Fact) {
```
Comment (regehr) [inline lib/Tool/CandidateMapUtils.cpp]: there are already two functions in Souper that do this, do we really need a third?

File: tools/souper.cpp [hunk 2]
```diff
@@ -44,24 +44,6 @@ OutputFilename("o", cl::desc("Override output filename"),
 static cl::opt<bool> StaticProfile("souper-static-profile", cl::init(false),
     cl::desc("Static profiling of Souper optimizations (default=false)"));
 
-static cl::opt<bool> PrintNeg("print-neg", cl::init(false),
```
Comment (regehr) [inline tools/souper.cpp]: hmm, what were these options doing in the code? @jubitaneja ?

Comment (zhengyangl) [reply tools/souper.cpp]: These are dead codes

File: lib/Tool/CandidateMapUtils.cpp (hunk 1 above)
Comment (zhengyangl) [reply lib/Tool/CandidateMapUtils.cpp]: Oops I’ll get it fixed

File: tools/souper.cpp (hunk 2 above)
Comment (regehr) [reply tools/souper.cpp]: yeah I was wondering how they got in there in the first place, so we can do a better job avoiding nasty stuff like this in the future

**Comment by regehr** [comment]
thanks-- is this ready to go?

**Comment by zhengyangl** [comment]
yes!

**Comment by regehr** [comment]
LGTM thanks!
great job with all the test cases!!!

**Pull request merged**